components/director/internal/domain/document/converter_test.go

```diff
@@ -1 +1,144 @@
-package document
+package document_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/kyma-incubator/compass/components/director/internal/domain/document"
+	"github.com/kyma-incubator/compass/components/director/internal/domain/document/automock"
+	"github.com/kyma-incubator/compass/components/director/internal/graphql"
+	"github.com/kyma-incubator/compass/components/director/internal/model"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestConverter_ToGraphQL(t *testing.T) {
+	// given
+	testCases := []struct {
+		Name     string
+		Input    *model.Document
+		Expected *graphql.Document
+	}{
+		{
+			Name:     "All properties given",
+			Input:    fixModelDocument("1", "foo"),
+			Expected: fixGQLDocument("foo"),
+		},
+		{
+			Name:     "Empty",
+			Input:    &model.Document{},
+			Expected: &graphql.Document{},
+		},
+		{
+			Name:     "Nil",
+			Input:    nil,
+			Expected: nil,
+		},
+	}
+
+	for i, testCase := range testCases {
+		t.Run(fmt.Sprintf("%d: %s", i, testCase.Name), func(t *testing.T) {
+			// when
+			frConv := &automock.FetchRequestConverter{}
+			if testCase.Input != nil {
+				frConv.On("ToGraphQL", testCase.Input.FetchRequest).Return(testCase.Expected.FetchRequest)
+			}
+			converter := document.NewConverter(frConv)
+			res := converter.ToGraphQL(testCase.Input)
+
+			// then
+			assert.Equal(t, testCase.Expected, res)
+		})
+	}
+}
+
+func TestConverter_MultipleToGraphQL(t *testing.T) {
+	// given
+	input := []*model.Document{
+		fixModelDocument("1", "foo"),
+		fixModelDocument("2", "bar"),
+		{},
+		nil,
+	}
+	expected := []*graphql.Document{
+		fixGQLDocument("foo"),
+		fixGQLDocument("bar"),
+		{},
+	}
+	var nilPointer *model.FetchRequest
+
+	// when
+	frConv := &automock.FetchRequestConverter{}
+	frConv.On("ToGraphQL", input[0].FetchRequest).Return(expected[0].FetchRequest)
+	frConv.On("ToGraphQL", nilPointer).Return(nil)
```

Comment:
You forget about checking if expectations were met, If I add
```
	frConv.On("WhereIsWally", nilPointer).Return(nil)
```
this test will pass. Please add:
```
defer frConv.AssertExpectations(t)
```

Reply:
Please check other places